src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTask.java

```diff
@@ -122,32 +113,8 @@ public void put(final Collection<SinkRecord> records) {
       LOGGER.debug("No sink records to process for current poll operation");
       return;
     }
-    Map<String, RecordBatches> batchMapping = createSinkRecordBatchesPerTopic(records);
```

Comment:
This logic has been simplified out into the `MongoProcessedSinkRecordData` & `MongoSinkRecordProcessor` classes.